src/rt/cpp/cown.h

```diff
@@ -0,0 +1,165 @@
+// Copyright Microsoft and Project Verona Contributors.
+// SPDX-License-Identifier: MIT
+
+#include <functional>
+#include <tuple>
+#include <utility>
+#include <verona.h>
+
+class cown_ptr_trait
+{
+private:
+  cown_ptr_trait() {}
+
+  template<typename T>
+  friend class cown_ptr;
+};
+
+/**
+ * Smart pointer to represent shared access to a cown.
+ * Can only be used asychronously with `when` to get
+ * underlying access.
+ */
+template<typename T>
+class cown_ptr : cown_ptr_trait
+{
+private:
+  /**
+   * Internal Verona runtime cown for this type.
+   */
+  class ActualCown : public VCown<ActualCown>

```

Comment:
It would be good to be consistent about naming.  Here the type starts with a capital (which, I believe, is the style that we've used everywhere else), but `cown_ptr` and `acquired_cown` use lower case and underscore separators.

Reply:
It's actually helpful when using LLVM containers/features that C++ didn't have in some previous standard but does have now. So, when we deprecate previous compilers (and bump the minimum standard), it's a lot easier to spot the LLVM ones from the standard ones.

I know it's not a very strong argument, but I'm also not sure we need to mimic the C++ standard naming here either.